packages/augur-core/source/contracts/BaseSimpleDex.sol

```diff
@@ -0,0 +1,192 @@
+    string constant public symbol = "SDE";
+
+    IAugur public augur;
+    address public token;
+
+    uint256 public tokenReserve;
+    uint256 public cashReserve;
+    uint256 public blockNumberLast;
+    uint256 public tokenPriceCumulativeLast;
+
+    event Mint(address indexed sender, uint256 tokenAmount, uint256 cashAmount);
+    event Burn(address indexed sender, uint256 tokenAmount, uint256 cashAmount, address indexed to);
+    event Swap(address indexed sender, address indexed tokenIn, uint256 amountIn, uint256 amountOut, address indexed to);
+    event Sync(uint256 tokenReserve, uint256 cashReserve);
+
+    function initializeInternal(address _augurAddress, address _token) internal {
+        endInitialization();
+        IAugur _augur = IAugur(_augurAddress);
+        augur = _augur;
+        token = _token;
+
+        initializeCashSender(_augur.lookup("DaiVat"), _augur.lookup("Cash"));
+        require(cash != IERC20(0));
+    }
+
+    function transferToken(address _to, uint256 _value) private;
+
+    function getTokenBalance() public returns (uint256);
+
+    function transferCash(address _to, uint _value) private {
+        cashTransfer(_to, _value);
+    }
+
+    function getCashBalance() public view returns (uint256) {
+        return cashBalance(address(this));
+    }
+
+    function update(uint256 _tokenBalance, uint256 _cashBalance) internal {
+        uint256 _blockNumber = block.number;
+        uint256 _blocksElapsed = _blockNumber - blockNumberLast;
+        uint256 _cashReserve = cashReserve;
+        uint256 _tokenReserve = tokenReserve;
+        if (_tokenReserve != 0 && _cashReserve != 0 && _blocksElapsed > 0) {
+            // cannot reasonably overflow unless the supply of Cash, REP, or ETH became several OOM larger
+            uint256 _priceCumulativeIncrease = _cashReserve.mul(10**18).mul(_blocksElapsed).div(_tokenReserve);
+            onUpdate(_blocksElapsed, _priceCumulativeIncrease);
+            tokenPriceCumulativeLast += _priceCumulativeIncrease;
+        }
+        tokenReserve = _tokenBalance;
+        cashReserve = _cashBalance;
+        blockNumberLast = _blockNumber;
+        emit Sync(_tokenBalance, _cashBalance);
+    }
+
+    function onUpdate(uint256 _blocksElapsed, uint256 _priceCumulativeIncrease) internal;
+
+    function publicMint(address _to) public nonReentrant returns (uint256 _liquidity) {
+        uint256 _tokenBalance = getTokenBalance();
+        uint256 _cashBalance = getCashBalance();
+        uint256 _tokenReserve = tokenReserve;
+        uint256 _cashReserve = cashReserve;
+        uint256 _totalSupply = totalSupply;
+        uint256 _tokenAmount = _tokenBalance.sub(_tokenReserve);
+        uint256 _cashAmount = _cashBalance.sub(_cashReserve);
+
+        _liquidity = _totalSupply == 0 ?
+            _tokenAmount.mul(_cashAmount).sqrt() :
+            (_tokenAmount.mul(_totalSupply) / _tokenReserve).min(_cashAmount.mul(_totalSupply) / _cashReserve);
```

Comment:
Uniswap has to support tokens with arbitrary scale, which is why they set liquidity to this rather complicated value (note the square root).  Since we only have to deal with tokens that are of the scale 10^18, and you can multiply 10^18 by itself about 4 times before you are in danger of running out of space in a uint256, we can instead have liquidity just be `k` (meaning `token_a * token_b`).  I believe such a change will simplify some calculations in a few places, including this one.

Reply:
From a usability standpoint, having the initial liquidity value be so large is likely to be awkward for things that treat it as an ERC20. Depositing 1 ETH and 160 DAI will yield 160e18 tokens. I think the sqrt is just a usability thing?